tests/cupy_tests/math_tests/test_sumprod.py

```diff
@@ -178,6 +178,182 @@ def test_prod_dtype(self, xp, src_dtype, dst_dtype):
         return a.prod(dtype=dst_dtype)
 
 
+@testing.gpu
+class TestNansumNanprod(unittest.TestCase):
```

Comment:
Could you add the test also for integer inputs? (The behavior will be expected to be same as `numpy.sum` and `numpy.prod`'s one.)

Reply:
Done.

Reply:
How about parameterizing some parameters (e.g., shapes, axis, Ellipsis...) with `cupy.testing.parameterize`?